src/main/java/com/cx/restclient/ast/AstClient.java

```diff
@@ -187,4 +207,55 @@ protected void handleInitError(IOException e) {
         String message = String.format("Failed to init %s client.", getScannerDisplayName());
         throw new CxClientException(message, e);
     }
+
+    protected HttpResponse submitAllSourcesFromLocalDir(String projectId, String zipFilePath) throws IOException {
+        log.info("Using local directory flow.");
+
+        PathFilter filter = new PathFilter(config.getOsaFolderExclusions(), config.getOsaFilterPattern(), log);
+        String sourceDir = config.getEffectiveSourceDirForDependencyScan();
+        File zipFile = CxZipUtils.getZippedSources(config, filter, sourceDir, log);
+
+        return initiateScanForUpload(projectId, zipFile, zipFilePath);
+    }
+
+    protected HttpResponse initiateScanForUpload(String projectId, File zipFile, String zipFilePath) throws IOException {
```

Comment:
Can we use zipFile.getPath().toString() here, so that the zipFilePath argument won't be needed?